gpu_ocean/SWESimulators/CDKLM16.py

```diff
@@ -275,6 +296,45 @@ def __init__(self, \
         self.angle_texref.set_address_mode(1, cuda.address_mode.CLAMP)
         self.angle_texref.set_flags(cuda.TRSF_NORMALIZED_COORDINATES) #Use [0, 1] indexing
         
+        
+        
+        # Texture for coriolis f
+        self.coriolis_texref = self.kernel.get_texref("coriolis_f_tex")
+        
+        # Create the CPU coriolis
+        if (latitude is not None):
+            if (self.f != 0.0):
+                raise RuntimeError("Cannot specify both latitude and f. Make your mind up.")
+            if (subsample_latitude and latitude.shape == eta0.shape):
+                self.logger.info("Subsampling latitude texture by factor " + str(subsample_latitude))
+                latitude = subsample(latitude, subsample_latitude)
+            coriolis_f, _ = OceanographicUtilities.calcCoriolisParams(latitude)
+            coriolis_f = coriolis_f.astype(np.float32)
+        else:
+            if (self.coriolis_beta != 0.0):
+                if (angle.size != 1):
+                    raise RuntimeError("non-constant angle cannot be combined with beta plane model (makes no sense)")
+                x = np.linspace(0, self.nx*self.dx, self.nx//subsample_f)
+                y = np.linspace(0, self.ny*self.dy, self.ny//subsample_f)
+                self.logger.info("Using latitude to create Coriolis f texture ({:f}x{:f} cells)".format(x.size, y.size))
+                x, y = np.meshgrid(x, y)
+                n = x*np.sin(angle[0, 0]) + y*np.cos(angle[0, 0])
+                coriolis_f = self.f + self.coriolis_beta*(n)
```

Comment:
Yes, that would be an improvement performance wise. I'll add a fixme on that for future development (requires a bit of thought into making sure that the coordinates are correct - and I don't have the time for that now)

Reply:
Actually, I changed my mind: Because of global ghost cells and texturing coordinates + linear interpolation it's not that simple.

Reply:
One possible way to implement would be to set the center of the first texel to correspond to 0.5dx, 0.5dy in the simulation, and equivalently the center of the last texel to correspond to nx-0.5dx, ny-0.5dy. This because the linear interpolation does not extend towards the boundary of the texture. Not implementing now, but left for later